src/common/iconv.c

```diff
@@ -67,18 +67,31 @@ int guac_iconv(guac_iconv_read* reader, const char** input, int in_remaining,
     while (in_remaining > 0 && out_remaining > 0) {
 
         int value;
+        int count;
         const char* read_start;
         char* write_start;
 
         /* Read character */
         read_start = *input;
         value = reader(input, in_remaining);
-        in_remaining -= *input - read_start;
+        count = *input - read_start;
+
+        /* Stop if read zero byte */
+        if (count == 0)
+            break;
+
+        in_remaining -= count;
 
         /* Write character */
         write_start = *output;
         writer(output, out_remaining, value);
-        out_remaining -= *output - write_start;
+        count = *output - write_start;
+
+        /* Stop if write zero byte */
```

Comment:
Before moving forward with this, can you clarify why writing zero bytes should universally be considered a failure condition?

It definitely makes sense to bail out if no bytes are read, as that will result in an infinite parse loop, but it's not necessarily illegal for a writer to choose not to write any output for a given input. The only case where that would *definitely* be considered a failure would be if `value` is 0, as `guac_iconv()` should always fail if it cannot write a null terminator.